parsers/GB_NIR.py

```diff
@@ -0,0 +1,245 @@
+"""
+Parser for Northern Ireland, uses the SONI / EirGrid SmartGrid dashboard. The
+following data types are available:
+
+Consumption
+Production
+Exchanges GB->GB-NIR and GB-NIR->IE
+"""
+
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from typing import Optional
+
+from dateutil import tz
+from requests import ConnectionError, Response, Session
+
+from parsers.lib.config import refetch_frequency
+from parsers.lib.exceptions import ParserException
+
+TZ = "Europe/Dublin"
+
+
+def _parse_effective_time(time_str: str):
+    """
+    Parses the EffectiveTime timestamps (e.g. 27-Dec-2022 00:00:00) from
+    SmartGrid dashboard into TZ-aware datetime objects.
+    """
+    naive_dt = datetime.strptime(time_str, "%d-%b-%Y %H:%M:%S")
+    return naive_dt.replace(tzinfo=tz.gettz(name=TZ))
+
+
+def _fetch_json_data(
+    region: str,
+    dataset: str,
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+) -> dict:
+    """
+    Fetches the requested json records for the specified date between 00:00 and
+    23:59 from the SmartGrid monitor.
+
+    Parameters:
+        region: name of the region, options "NI", "ROI" or "ALL".
+
+        dataset: name of the dataset, e.g. demandactual, interconnection etc.
+        Examine the SmartGrid dashboard for available options.
+
+    Returns:
+        The fetched rows of data as a list of dictionaries.
+    """
+    DATA_URL = "https://www.smartgriddashboard.com/DashboardService.svc/data"
+    try:
+        if not target_datetime:
+            target_datetime = datetime.now(tz.gettz(name=TZ))
+
+        # get all available data for a single day
+        date_from = target_datetime.replace(hour=0, minute=0, second=0)
+        date_to = target_datetime.replace(hour=23, minute=59, second=59)
+
+        params = {
+            "area": dataset,
+            "region": region,
+            "datefrom": date_from,
+            "dateto": date_to,
+        }
+
+        response: Response = session.get(DATA_URL, params=params)
+
+        return response.json()["Rows"]
+    except ConnectionError as e:
+        ParserException("GB_NIR.py", f"Failed to connect to SmartGrid Dashboard: {e}")
+        raise ConnectionError
+
+
+@refetch_frequency(timedelta(minutes=15))
```

Comment:
```suggestion
@refetch_frequency(timedelta(days=2))
```
With the above change this becomes 2 days.

Reply:
One question, does this push the data to be older than the expected 2 hours? The SONI dashboard and the json data source updates every 15 minutes during the day but would this imply we pull their data only every 2 days?